internal/services/sentinel/sentinel_automation_rule_resource_test.go

```diff
@@ -180,6 +239,213 @@ resource "azurerm_sentinel_automation_rule" "test" {
+  display_name               = "acctest-SentinelAutoRule-%d-update"
+  order                      = 2
+  enabled                    = false
+  expiration                 = "2022-11-20T15:44:52Z"
+
+  condition_json = jsonencode(
+    [
+      {
+        conditionProperties = {
+          operator     = "Contains"
+          propertyName = "IncidentTitle"
+          propertyValues = [
+            "a",
+            "b",
+          ]
+        }
+        conditionType = "Property"
+      },
+      {
+        conditionProperties = {
+          operator     = "Contains"
+          propertyName = "IncidentTitle"
+          propertyValues = [
+            "c",
+            "d",
+          ]
+        }
+        conditionType = "Property"
+      },
+    ]
+  )
+
+  action_incident {
+    order                  = 1
+    status                 = "Closed"
+    classification         = "BenignPositive_SuspiciousButExpected"
+    classification_comment = "whatever reason"
+  }
+
+  action_incident {
+    order  = 3
+    labels = ["foo", "bar"]
+  }
+
+  action_incident {
+    order    = 2
+    severity = "High"
+  }
+
+  action_incident {
+    order    = 4
+    owner_id = data.azurerm_client_config.current.object_id
+  }
+}
+`, template, r.uuid, data.RandomInteger)
+}
+
+func (r SentinelAutomationRuleResource) triggerIncidentUpdated(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_sentinel_automation_rule" "test" {
+  name                       = "%s"
+  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
+  display_name               = "acctest-SentinelAutoRule-%d-update"
+  order                      = 1
+  condition_json = jsonencode([
+    {
+      conditionType = "PropertyChanged"
+      conditionProperties = {
+        propertyName   = "IncidentStatus"
+        changeType     = "ChangedTo"
+        operator       = "Equals"
+        propertyValues = ["New"]
+      }
+    }
+  ])
+
+  triggers_when = "Updated"
+
+  action_incident {
+    order    = 1
+    owner_id = data.azurerm_client_config.current.object_id
+  }
+}
+`, template, r.uuid, data.RandomInteger)
+}
+
+func (r SentinelAutomationRuleResource) triggerAlertCreated(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+data "azurerm_client_config" "current" {}
+
+data "azurerm_managed_api" "test" {
+  name     = "azuresentinel"
+  location = azurerm_resource_group.test.location
+}
+
+resource "azurerm_api_connection" "test" {
+  managed_api_id      = data.azurerm_managed_api.test.id
+  name                = "azuresentinel-%[2]d"
+  resource_group_name = azurerm_resource_group.test.name
+  parameter_values = {
+    "token:TenantId"     = data.azurerm_client_config.current.tenant_id
+    "token:clientId"     = data.azurerm_client_config.current.client_id
+    "token:clientSecret" = %[4]q
```

Comment:
we can't upload the client secret here, since this isn't marked as sensitive.. can this use MSI instead?

Reply:
@tombuildsstuff I've verified that the returned response won't contain the secret:

```
{
  "properties": {
        ...
        "nonSecretParameterValues": {
            "token:TenantId": "xxxx",
            "token:clientId": "xxxx",
            "token:grantType": "client_credentials"
        },
        ...
}
```

Reply:
Regarding MSI, the Portal can create such an api connection, whose API request is as below:

```json
{
    "properties": {
        "api": {
            "id": "/subscriptions/xxxx/providers/Microsoft.Web/locations/eastus2/managedApis/azuresentinel"
        },
        "parameterValueType": "Alternative",
        "alternativeParameterValues": {},
        "displayName": "test"
    },
    "kind": "V1",
    "location": "eastus2"
}
```

The `parameterValueType` should be the important one, but it is not defined in the Swagger for version 2016-06-01 (which we are using in the provider). Instead, above API is using 
2018-07-01-preview, but this version isn't defined in the Swagger repo...

Reply:
👍